src/Nethermind/Nethermind.Optimism/CL/ExecutionEngineManager.cs

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Threading.Tasks;
+using Nethermind.Core.Crypto;
+using Nethermind.Logging;
+using Nethermind.Merge.Plugin.Data;
+using Nethermind.Optimism.CL.Derivation;
+
+namespace Nethermind.Optimism.CL;
+
+public class ExecutionEngineManager : IExecutionEngineManager
+{
+    private readonly IL2Api _l2Api;
+    private readonly IDerivedBlocksVerifier _derivedBlocksVerifier;
+    private readonly ILogger _logger;
+
+    private BlockId _currentHead;
+    private BlockId _currentFinalizedHead;
+    private BlockId _currentSafeHead;
+
+    public ExecutionEngineManager(IL2Api l2Api, ILogger logger)
+    {
+        _l2Api = l2Api;
+        _logger = logger;
+        _derivedBlocksVerifier = new DerivedBlocksVerifier(logger);
+    }
+
+    public async Task Initialize()
+    {
+        var headBlock = await _l2Api.GetHeadBlock();
+        _currentHead = BlockId.FromL2Block(headBlock);
+
+        var finalizedBlock = await _l2Api.GetFinalizedBlock();
+        _currentFinalizedHead = BlockId.FromL2Block(finalizedBlock);
+
+        var safeBlock = await _l2Api.GetSafeBlock();
+        _currentSafeHead = BlockId.FromL2Block(safeBlock);
```

Comment:
We can run these Tasks concurrently instead of awaiting each one sequentially.

```suggestion
        var headBlock = _l2Api.GetHeadBlock();  
        var finalizedBlock = _l2Api.GetFinalizedBlock();  
        var safeBlock = _l2Api.GetSafeBlock();  

        _currentHead = BlockId.FromL2Block(await headBlock);  
        _currentFinalized = BlockId.FromL2Block(await finalizedBlock);  
        _currentSafe = BlockId.FromL2Block(await safeBlock);  
```

Reply:
This is initialization code, let's keep it as simple as possible.